Sources/HummingbirdCore/Request/RequestBody+inboundClose.swift

```diff
@@ -0,0 +1,144 @@
+
+import HTTPTypes
+import NIOConcurrencyHelpers
+import NIOCore
+import NIOHTTPTypes
+
+#if compiler(>=6.0)
+extension RequestBody {
+    /// Run provided closure but cancel it if the inbound request part stream is closed.
+    ///
+    /// This function is designed for use with long running requests like server sent events. It assumes you
+    /// are not going to be using the request body after calling as it consumes the request body, it also assumes
+    /// you havent edited the request body prior to calling this function.
+    ///
+    /// If the response finishes the connection will be closed.
+    ///
+    /// - Parameters
+    ///   - isolation: The isolation of the method. Defaults to the isolation of the caller.
+    ///   - operation: The actual operation
+    ///   = onInboundClose: handler invoked when inbound is closed
+    /// - Returns: Return value of operation
+    public func consumeWithInboundCloseHandler<Value: Sendable>(
+        isolation: isolated (any Actor)? = #isolation,
+        _ operation: (RequestBody) async throws -> Value,
+        onInboundClosed: @Sendable @escaping () -> Void
+    ) async throws -> Value {
+        let iterator: UnsafeTransfer<NIOAsyncChannelInboundStream<HTTPRequestPart>.AsyncIterator> =
+            switch self._backing {
+            case .nioAsyncChannelRequestBody(let iterator):
+                iterator.underlyingIterator
+            default:
+                preconditionFailure("Cannot run consumeWithInboundCloseHandler on edited request body")
+            }
+        let (requestBody, source) = RequestBody.makeStream()
+        return try await withInboundCloseHandler(
+            iterator: iterator.wrappedValue,
+            source: source,
+            operation: {
+                try await operation(requestBody)
+            },
+            onInboundClosed: onInboundClosed
+        )
+    }
+
+    /// Run provided closure but cancel it if the inbound request part stream is closed.
+    ///
+    /// This function is designed for use with long running requests like server sent events. It assumes you
+    /// are not going to be using the request body after calling as it consumes the request body, it also assumes
+    /// you havent edited the request body prior to calling this function.
+    ///
+    /// If the response finishes the connection will be closed.
+    ///
+    /// - Parameters
+    ///   - isolation: The isolation of the method. Defaults to the isolation of the caller.
+    ///   - operation: The actual operation to run
+    /// - Returns: Return value of operation
+    public func consumeWithCancellationOnInboundClose<Value: Sendable>(
+        _ operation: sending @escaping (RequestBody) async throws -> Value
+    ) async throws -> Value {
+        let (barrier, source) = AsyncStream<Void>.makeStream()
+        return try await consumeWithInboundCloseHandler { body in
+            try await withThrowingTaskGroup(of: Value.self) { group in
```

Comment:
Why though? sure, `~Escapable` would give us compile-time blessing, but if you don't pass the group anywhere, it will not escape. I don't see how these are connected, really.

Reply:
I agree with @sliemeobn here, we should consider `withoutActuallyEscaping`